### PR TITLE
CLDR-17287 refactor TestSTFactory to JUnit

### PR DESCRIPTION
**Pull request by srl295**
- move from TestFmwk to JUnit
- fix concurrency, simplify test cases
- [ ] split out data driven reader

CLDR-12787

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by srl295** [comment]
implements refactoring suggested in #3418

File: tools/cldr-apps/src/test/java/org/unicode/cldr/web/TestSTFactory.java [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+        runDataDrivenTest("TestVotePerf");
+    }
+
+    public void TestUserRegistry() throws SQLException, IOException {
+        runDataDrivenTest("TestUserRegistry");
+    }
+
+    /*
+     * TODO: shorten this function, over 300 lines; use subroutines
+     */
```
Comment (btangmu) [inline tools/cldr-apps/src/test/java/org/unicode/cldr/web/TestSTFactory.java:478]: this comment can be removed now :-)

**Review by btangmu — APPROVED** [review]

File: tools/cldr-apps/src/test/java/org/unicode/cldr/web/TestSTFactory.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/test/java/org/unicode/cldr/web/TestSTFactory.java]: gone!

**Review by btangmu — APPROVED** [review]

**Comment by btangmu** [comment]
@srl295 I edited the description to change the ticket from CLDR-12787 to CLDR-17287

**Pull request merged**